custom/icds_reports/static/js/directives/underweight-children-report/underweight-children-report.directive.js

```diff
@@ -80,9 +80,17 @@ function UnderweightChildrenReportController($scope, $routeParams, $location, $f
         vm.setStepsMapLabel();
         var usePercentage = true;
         var forceYAxisFromZero = false;
-        vm.myPromise = maternalChildService.getUnderweightChildrenData(vm.step, vm.filtersData).then(
-            vm.loadDataFromResponse(usePercentage, forceYAxisFromZero)
-        );
+        // mobile dashboard requires all data on both pages, whereas web just requires the current step's data
+        // note: it would be better to not load this data on both step pages but instead save it in the JS, but
+        // doing that now would be a bit complicated and the server-side caching should make the switching
+        // relatively painless
```

Comment:
This comment is probably worth reading. I explored being able to not rely on the angular routing stuff for navigation between the steps but decided that was too much to bite off and that I was ok living with this.